packages/block-editor/src/components/link-control/use-internal-input-value.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+
+export default function useInternalInputValue( value ) {
```

Comment:
I wonder if we already have a similar hook (or maybe it's something from reakit). Anyway @ciampo or @diegohaz would know

Reply:
Would using `key` to force remount the component be simpler in this context?

Reply:
for a use case like a different link in a RichText component, it potentially makes sense to force remount but it's a fairly common practice in reusable components to have "internal/external state and changes in external forcing updates to internal state"

Reply:
> t's a fairly common practice in reusable components to have "internal/external state and changes in external forcing updates to internal state"

I'm not sure, it's essentially the same as a derived state, which [isn't really common](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html) according to React core team. And seems like we can just [call `setState` during the rendering phase](https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops) if we really need to do this (which I'm not sure if is still the recommended approach in concurrent rendering 🤔 ).

Reply:
Yeh I always felt a little uncomfortable regarding this pattern here.

The key UX here is that the `value` shouldn't be committed until the user takes an explicit action (e.g. hitting `Submit` or `ENTER`). Until that point the data the user has typed should be considered local to the component because if the user clicks away we should be able to discard the change.

That is why the internal state exists.

Reply:
When we use `LinkControl` we now often pass a `key` prop to force re-render the component when a "new" link is being handled (like when you move between links in a paragraph block).